sklearn/preprocessing/_encoders.py

```diff
@@ -371,20 +475,34 @@ def _compute_drop_idx(self):
                                             len(self.drop)))
             missing_drops = []
             drop_indices = []
+
+            def _convert_to_infrequent_idx(idx, col_idx):
```

Comment:
It might be nice to add a docstring on the purpose of the function

Reply:
I don't know if we should put it outside of `_compute_drop_idx`

Reply:
Moved it outside of `_compute_drop_idx`.